Refactor and 1.17-pre1

shedaniel opened the pull request:
- New package structure:
![image](https://user-images.githubusercontent.com/34910653/120066767-98796d00-c0aa-11eb-926a-420d122ba60a.png)

- Removed deprecated apis
- Relocated to `dev.architectury`
- Update to architectury plugin 3.2, which uses `dev.architectury` @ExpectPlatform
- Migrate all InteractionResult to EventResult

shedaniel committed:
Fix #94

Signed-off-by: shedaniel <daniel@shedaniel.me>

canitzp committed:
Add Chunk data save and load events, closes #89 (#92)

* Created Chunk save and load event. Closes #89

* Fixed missing licence

* Update common/src/main/java/me/shedaniel/architectury/event/events/ChunkEvent.java

* Update common/src/main/java/me/shedaniel/architectury/event/events/ChunkEvent.java

* Update common/src/main/java/me/shedaniel/architectury/event/events/ChunkEvent.java

* Supply ServerLevel in ChunkEvent.LOAD, style cleanup

Signed-off-by: shedaniel <daniel@shedaniel.me>

* Add "Data" suffix to Chunk IO Events and mark level as nullable for load

* Update common/src/main/java/me/shedaniel/architectury/event/events/ChunkEvent.java

* Bump to 1.16

Co-authored-by: shedaniel <daniel@shedaniel.me>
Co-authored-by: Max <maxh2709@gmail.com>

MaxNeedsSnacks committed:
Update pr.yml

canitzp committed:
Add NetworkManager hook for Entity spawn packets (#88)

* New utility hook for creating a entity spawn packet. before every mod had to implement this in itself.

* Apply suggestions from code review

Co-authored-by: shedaniel <daniel@shedaniel.me>

* Update common/src/main/java/me/shedaniel/architectury/networking/NetworkManager.java

* Update common/src/main/java/me/shedaniel/architectury/networking/NetworkManager.java

* Properly implement SpawnEntityPacket & Format style

Signed-off-by: shedaniel <daniel@shedaniel.me>

* Format

Signed-off-by: shedaniel <daniel@shedaniel.me>

* createEntitySpawnPacket -> createAddEntityPacket for mojmap consistency

Signed-off-by: shedaniel <daniel@shedaniel.me>

* Bump version to 1.17

Co-authored-by: Max <maxh2709@gmail.com>
Co-authored-by: shedaniel <daniel@shedaniel.me>

canitzp committed:
Add Javadocs for all events (#91)

* javadoc for ClientChatEvent

* javadoc for ClientPlayerEvent

* javadoc for ClientRawInputEvent

* Added javadoc for ClientScreenInputEvent

* Bit refactoring and removed javadoc @see to the caller methods

* Add javadoc for BlockEvent

* Add javadoc for ChatEvent

* Add javadoc for CommandPerformEvent and CommandRegistrationEvent

* Add javadoc for EntityEvent

* Add javadoc for ExplosionEvent

* Add javadoc for GuiEvent

* Add javadoc for InteractionEvent

* Add javadoc for LightningEvent, RecipeUpdateEvent & TextureStitchEvent

* Add javadoc for LifecycleEvent & ClientLifecycleEvent

* Add javadoc for TooltipEvent

* Add javadoc for TickEvent and partly PlayerEvent

* More javadoc for PlayerEvent

* Finally all event have javadoc

* Update common/src/main/java/me/shedaniel/architectury/event/events/CommandPerformEvent.java

Co-authored-by: Juuxel <6596629+Juuxel@users.noreply.github.com>

* Update common/src/main/java/me/shedaniel/architectury/event/events/CommandPerformEvent.java

Co-authored-by: Juuxel <6596629+Juuxel@users.noreply.github.com>

* Update common/src/main/java/me/shedaniel/architectury/event/events/CommandPerformEvent.java

Co-authored-by: Juuxel <6596629+Juuxel@users.noreply.github.com>

* Update common/src/main/java/me/shedaniel/architectury/event/events/ChatEvent.java

Co-authored-by: Juuxel <6596629+Juuxel@users.noreply.github.com>

* Update common/src/main/java/me/shedaniel/architectury/event/events/CommandRegistrationEvent.java

Co-authored-by: Juuxel <6596629+Juuxel@users.noreply.github.com>

* Update common/src/main/java/me/shedaniel/architectury/event/events/EntityEvent.java

Co-authored-by: Juuxel <6596629+Juuxel@users.noreply.github.com>

* Update common/src/main/java/me/shedaniel/architectury/event/events/EntityEvent.java

Co-authored-by: Juuxel <6596629+Juuxel@users.noreply.github.com>

* Update common/src/main/java/me/shedaniel/architectury/event/events/EntityEvent.java

Co-authored-by: Juuxel <6596629+Juuxel@users.noreply.github.com>

* Update common/src/main/java/me/shedaniel/architectury/event/events/EntityEvent.java

Co-authored-by: Juuxel <6596629+Juuxel@users.noreply.github.com>

* Update common/src/main/java/me/shedaniel/architectury/event/events/client/ClientScreenInputEvent.java

Co-authored-by: Juuxel <6596629+Juuxel@users.noreply.github.com>

* Update common/src/main/java/me/shedaniel/architectury/event/events/client/ClientScreenInputEvent.java

Co-authored-by: Juuxel <6596629+Juuxel@users.noreply.github.com>

* Update common/src/main/java/me/shedaniel/architectury/event/events/BlockEvent.java

Co-authored-by: Max <maxh2709@gmail.com>

* Update common/src/main/java/me/shedaniel/architectury/event/events/BlockEvent.java

Co-authored-by: Max <maxh2709@gmail.com>

* Update common/src/main/java/me/shedaniel/architectury/event/events/BlockEvent.java

Co-authored-by: Max <maxh2709@gmail.com>

* Update common/src/main/java/me/shedaniel/architectury/event/events/BlockEvent.java

Co-authored-by: Max <maxh2709@gmail.com>

* Update common/src/main/java/me/shedaniel/architectury/event/events/LifecycleEvent.java

Co-authored-by: Max <maxh2709@gmail.com>

* Update common/src/main/java/me/shedaniel/architectury/event/events/LifecycleEvent.java

Co-authored-by: Max <maxh2709@gmail.com>

* Update common/src/main/java/me/shedaniel/architectury/event/events/LifecycleEvent.java

Co-authored-by: Max <maxh2709@gmail.com>

* Update common/src/main/java/me/shedaniel/architectury/event/events/PlayerEvent.java

Co-authored-by: Max <maxh2709@gmail.com>

* Apply suggestions from code review

Co-authored-by: Max <maxh2709@gmail.com>
Co-authored-by: Juuxel <6596629+Juuxel@users.noreply.github.com>

* Apply suggestions from code review

Co-authored-by: Juuxel <6596629+Juuxel@users.noreply.github.com>
Co-authored-by: Max <maxh2709@gmail.com>

* Apply suggestions from code review

Co-authored-by: Max <maxh2709@gmail.com>
Co-authored-by: Juuxel <6596629+Juuxel@users.noreply.github.com>

* Apply suggestions from code review

Co-authored-by: Max <maxh2709@gmail.com>
Co-authored-by: Juuxel <6596629+Juuxel@users.noreply.github.com>

* Apply suggestions from code review

Co-authored-by: Max <maxh2709@gmail.com>
Co-authored-by: Juuxel <6596629+Juuxel@users.noreply.github.com>

* Apply suggestions from code review

* Apply suggestions from code review

Co-authored-by: shedaniel <daniel@shedaniel.me>

* Applied some more reviews

* Applied some more reviews

* Applied suggestions

* Reformat and apply some minor suggestions

* Apply suggestions from review (first batch)

* Remove double return javadoc

* Apply suggestions from code review

Co-authored-by: shedaniel <daniel@shedaniel.me>

* Add "scaled" to mouse coordinates

* Final touches

Signed-off-by: shedaniel <daniel@shedaniel.me>

* Remove invalid tips

Signed-off-by: shedaniel <daniel@shedaniel.me>

Co-authored-by: Juuxel <6596629+Juuxel@users.noreply.github.com>
Co-authored-by: Max <maxh2709@gmail.com>
Co-authored-by: shedaniel <daniel@shedaniel.me>

shedaniel committed:
Merge remote-tracking branch 'architectury/1.16' into 1.17

# Conflicts:
#	.github/workflows/dev_pr.yml
#	common/src/main/java/me/shedaniel/architectury/event/events/ChatEvent.java
#	common/src/main/java/me/shedaniel/architectury/event/events/EntityEvent.java
#	common/src/main/java/me/shedaniel/architectury/event/events/GuiEvent.java
#	common/src/main/java/me/shedaniel/architectury/event/events/PlayerEvent.java
#	fabric/src/main/resources/architectury.mixins.json
#	gradle.properties

shedaniel committed:
Refactor

Signed-off-by: shedaniel <daniel@shedaniel.me>

MaxNeedsSnacks commented:
This is getting an immediate "last call" because it's **critical** for 1.17 project structure, so please, if you have any concerns regarding this pull request, voice them now!

MaxNeedsSnacks commented on common/src/main/java/dev/architectury/event/EventFactory.java:
```diff
@@ -63,28 +61,6 @@ protected Object handleInvocation(@NotNull Object proxy, @NotNull Method method,
     }
     
-    @SafeVarargs
-    public static <T> Event<T> createInteractionResult(T... typeGetter) {
```
We should mark these as `@Deprecated` in 1.x to let people creating their own events know they are being removed in favour of `EventResult`s

MaxNeedsSnacks commented on common/src/main/java/dev/architectury/event/events/common/EntityEvent.java:
```diff
@@ -0,0 +1,133 @@
+    /**
+     * @see LivingAttack#attack(LivingEntity, DamageSource, float)
+     */
+    Event<LivingAttack> LIVING_ATTACK = EventFactory.createEventResult();
```
I've been contemplating whether we should change this to `LIVING_DAMAGE(D)` as that is a bit more descriptive of what's actually happening here (the living entity is taking **damage** after *being attacked*, as opposed to *attacking* something actively)

MaxNeedsSnacks replied:
For reference, the vanilla method is `LivingEntity#hurt`, so maybe `LIVING_HURT` might be more fitting

MaxNeedsSnacks commented on common/src/main/java/dev/architectury/event/events/common/InteractionEvent.java:
```diff
@@ -0,0 +1,160 @@
+         * @param pos    The position of the block in the level.
+         * @param face   The face of the block clicked.
+         * @return A {@link EventResult} determining the outcome of the event,
+         * the execution of the vanilla click may be cancelled by the result.
```
Could be shortened to just
```suggestion
         * the action may be cancelled by the result.
```

MaxNeedsSnacks commented on common/src/main/java/dev/architectury/event/events/common/InteractionEvent.java:
```diff
@@ -0,0 +1,160 @@
+         * @param pos    The position of the block in the level. Use {@link Player#getCommandSenderWorld()} to get the level.
+         * @param face   The face of the block clicked.
+         * @return A {@link EventResult} determining the outcome of the event,
+         * the execution of the vanilla click may be cancelled by the result.
```
Could be shortened to just
```suggestion
         * the action may be cancelled by the result.
```

MaxNeedsSnacks commented on common/src/main/java/dev/architectury/event/events/common/InteractionEvent.java:
```diff
@@ -0,0 +1,160 @@
+         * @param player The player right clicking the block.
+         * @param hand   The hand that is used.
+         * @return A {@link EventResult} determining the outcome of the event,
+         * the execution of the vanilla click may be cancelled by the result.
```
Could be shortened to just
```suggestion
         * the action may be cancelled by the result.
```

MaxNeedsSnacks commented on common/src/main/java/dev/architectury/event/events/common/InteractionEvent.java:
```diff
@@ -0,0 +1,160 @@
+         * @param entity Then entity the player clicks.
+         * @param hand   The used hand.
+         * @return A {@link EventResult} determining the outcome of the event,
+         * the execution of the vanilla interaction may be cancelled by the result.
```
Could be shortened to just
```suggestion
         * the action may be cancelled by the result.
```

MaxNeedsSnacks commented on common/src/main/java/dev/architectury/event/events/common/InteractionEvent.java:
```diff
@@ -0,0 +1,160 @@
+         * @param state    The state of the block.
+         * @param distance The distance of the player to the block.
+         * @param entity   The entity trampling.
+         * @return The event callback result.
```
Debating whether this could use the same detail as the events above

shedaniel committed:
Update common/src/main/java/dev/architectury/event/events/common/InteractionEvent.java

Co-authored-by: Max <maxh2709@gmail.com>

MaxNeedsSnacks commented on common/src/main/java/dev/architectury/event/events/common/LifecycleEvent.java:
```diff
@@ -0,0 +1,115 @@
+    interface ServerState extends InstanceState<MinecraftServer> {
+    }
+    
+    interface WorldState<T extends Level> {
```
General question: Should we keep using `World` to refer to `Level`? (as Forge is likely switching to Mojmap anyways and I doubt this counts as "mappings taint" for Yarn users due to how ubiquitous this is)

shedaniel committed:
Apply suggestions from code review

Co-authored-by: Max <maxh2709@gmail.com>

MaxNeedsSnacks commented on common/src/main/java/dev/architectury/registry/level/entity/trade/SimpleTrade.java:
```diff
@@ -37,14 +37,9 @@
-    private final int experiencePoints;
-    private final float priceMultiplier;
-    
+public record SimpleTrade(ItemStack primaryPrice, ItemStack secondaryPrice,
```
Good catch! Might be worth Java 16-ifying the entire project a bit later down the line

shedaniel committed:
attack -> hurt

Signed-off-by: shedaniel <daniel@shedaniel.me>

shedaniel committed:
Update @param for InteractionEvent#FARMLAND_TRAMPLE

Signed-off-by: shedaniel <daniel@shedaniel.me>

shedaniel committed:
world -> level

Signed-off-by: shedaniel <daniel@shedaniel.me>

MaxNeedsSnacks commented:
Bit busy with external work right now, so if someone could do a general second pass to make sure the formatting's alright and the overall API structure makes sense, that'd be great ^^

shedaniel committed:
Relocate hooks

Signed-off-by: shedaniel <daniel@shedaniel.me>

shedaniel committed:
Organize imports

Signed-off-by: shedaniel <daniel@shedaniel.me>

shedaniel committed:
Registry -> Registrar Please voice out your opinion!!

Signed-off-by: shedaniel <daniel@shedaniel.me>

shedaniel committed:
Some more renames
itemOptional -> optionalItem
RegistryBuilder -> RegistrarBuilder
RegistryOption -> RegistrarOption
Not forcing RegistryEntry on new registries, vanilla types don't have them

Signed-off-by: shedaniel <daniel@shedaniel.me>

shedaniel committed:
Java 16-ification

Signed-off-by: shedaniel <daniel@shedaniel.me>

shedaniel committed:
Optimise imports after the var change

Signed-off-by: shedaniel <daniel@shedaniel.me>

shedaniel committed:
Un-minimize architectury-common.mixins.json

Signed-off-by: shedaniel <daniel@shedaniel.me>
